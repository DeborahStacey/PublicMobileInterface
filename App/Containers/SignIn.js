import React from 'react'
import { ScrollView, Text, Image, View, TextInput } from 'react-native'
import { Images } from '../Themes'
import RoundedButton from '../Components/RoundedButton'
// import { Actions as NavigationActions } from 'react-native-router-flux'
import {create} from 'apisauce'

// Styles
import styles from './Styles/PresentationScreenStyle'

const db = create({
  baseURL: 'https://cat.ddns.net/Backend/api.php',
  headers: {'Content-Type': 'application/json'}
})

export default class SignIn extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  signInUser () {
    if (this.state.email === '' || this.state.password === '') {
      window.alert('not all of the fields have been filled out please enter all data!')
      return
    }

    var postObj = {
      'email': this.state.email,
      'password': this.state.password
    }
    db.post('/user/login', postObj).then((response) => window.alert(JSON.stringify(response.data)))
  }

  updateEmail (event) {
    this.setState({
      email: event
    })
  }

  updatePassword (event) {
    this.setState({
      password: event
    })
  }

  focusNextField = (nextField) => {
    this.refs[nextField].focus()
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <Image source={Images.background} style={styles.backgroundImage} resizeMode='stretch' />
        <ScrollView style={styles.container}>

          <View style={styles.messageBox}>
            <View style={styles.messageBoxContents}>

              <Text style={styles.sectionText}>Email</Text>
              <TextInput ref="1" onSubmitEditing={() => this.focusNextField('2')} returnKeyType="next" onChangeText={this.updateEmail.bind(this)} placeholder='Enter your email' placeholderTextColor='white' style={styles.sectionInput} keyboardType="email-address" />

              <Text style={styles.sectionText}>Password</Text>
              <TextInput ref="2" onSubmitEditing={this.signInUser.bind(this)} returnKeyType="done" onChangeText={this.updatePassword.bind(this)} placeholder='Enter your password' placeholderTextColor='white' style={styles.sectionInput} secureTextEntry={true} />

            </View>
          </View>

          <RoundedButton onPress={this.signInUser.bind(this)}>
            Sign In
          </RoundedButton>

        </ScrollView>
      </View>
    )
  }
}
