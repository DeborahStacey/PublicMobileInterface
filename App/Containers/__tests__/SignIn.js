import 'react-native'
import React from 'react'
import SignIn from '../SignIn'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <SignIn />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
