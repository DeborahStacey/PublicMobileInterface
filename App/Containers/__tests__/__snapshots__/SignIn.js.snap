exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <Image
    resizeMode="stretch"
    source={1}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    } />
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Email
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your email"
          placeholderTextColor="white"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your password"
          placeholderTextColor="white"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e73536",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        SIGN IN
      </Text>
    </View>
  </ScrollView>
</View>
`;
